@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -222,12 +226,30 @@ export class TileGeometryCreator {
             }
             return filter(technique);
         };
+        this.createTextElements(tile, decodedTile, textFilter);
 
-        this.preparePois(tile, decodedTile);
+        this.createLabelRejectionElements(tile, decodedTile);
+    }
 
-        // TextElements do not get their geometry created by Tile, but are managed on a
-        // higher level.
-        this.createTextElements(tile, decodedTile, textFilter);
+    createLabelRejectionElements(tile: Tile, decodedTile: DecodedTile) {
+        if (decodedTile.pathGeometries === undefined) {
+            return;
+        }
+        const path: THREE.Vector3[] = [];
+        const point = new THREE.Vector3();
+        for (const flatpath of decodedTile.pathGeometries) {
+            for (let i = 0; i < flatpath.path.length; i += 3) {
+                point.set(flatpath.path[i], flatpath.path[i + 1], flatpath.path[i + 2]);
+                // Move to world coordinates, the accuracy isn't important, because we don't need to
+                // worry about z flickering (the path is not rendered). This also makes the code
+                // projecting to screen space in
+                // [[TextElementsRenderer.prepopulateScreenWithBlockingElements]] simpler because
+                // the shift before projection to NDC isn't required.
```

Comment:
@ninok , is this comment correct?

Reply:
The actual reason why we can use world coordinates for that is that we use double precision (b/c all the label stuff is done on the CPU).

Reply:
but what does double precision give us? More accuracy to avoid z fighting, or something else?